pkg/tracker/gas.go

```diff
@@ -36,7 +37,7 @@ func (b *GasTracker) String() string {
 	return "GasTracker"
 }
 
-func (b *GasTracker) Exec(ctx context.Context) error {
+func (b *GasTracker) Exec(ctx context.Context, logger log.Logger) error {
```

Comment:
Here we also need a constructor which takes a logger so that can have the component name in the log lines.

The same for all other places.